shared/src/main/java/uk/gov/di/authentication/shared/jwt/KmsJwsSigner.java

```diff
@@ -0,0 +1,63 @@
+package uk.gov.di.authentication.shared.jwt;
+
+import com.nimbusds.jose.JOSEException;
+import com.nimbusds.jose.JWSAlgorithm;
+import com.nimbusds.jose.JWSHeader;
+import com.nimbusds.jose.JWSSigner;
+import com.nimbusds.jose.crypto.impl.ECDSA;
+import com.nimbusds.jose.jca.JCAContext;
+import com.nimbusds.jose.util.Base64URL;
+import uk.gov.di.authentication.shared.services.KmsConnectionService;
+
+import java.util.Set;
+
+import static software.amazon.awssdk.services.kms.model.SigningAlgorithmSpec.ECDSA_SHA_256;
+import static software.amazon.awssdk.services.kms.model.SigningAlgorithmSpec.RSASSA_PKCS1_V1_5_SHA_256;
+
+public class KmsJwsSigner implements JWSSigner {
+
+    private final KmsConnectionService kms;
+    private final String rsaSigningKeyId;
+    private final String ecSigningKeyId;
+
+    public KmsJwsSigner(KmsConnectionService kms, String rsaSigningKeyId, String ecSigningKeyId) {
+        this.kms = kms;
+        this.rsaSigningKeyId = rsaSigningKeyId;
+        this.ecSigningKeyId = ecSigningKeyId;
+    }
+
+    @Override
+    public Base64URL sign(JWSHeader header, byte[] signingInput) throws JOSEException {
+        if (signingInput.length == 0) {
+            return new Base64URL("");
+        }
+
+        var signingKey =
+                header.getAlgorithm() == JWSAlgorithm.RS256 ? rsaSigningKeyId : ecSigningKeyId;
```

Comment:
We're making the `header.getAlgorithm() == JWSAlgorithm.RS256` quite a few times here. Perhaps something like

```suggestion

        if (header.getAlgorithm() == JWSAlgorithm.RS256) {
            return signRsa(byte[] signingInput);
        }
        return signEcdsa(byte[] signingInput);
```
?

Reply:
Yep, that makes sense.